delta-lake/delta-spark350db143/src/main/scala/com/nvidia/spark/rapids/delta/shims/UpdateCommandMetaShim.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.delta.shims
+
+import com.databricks.sql.transaction.tahoe.commands.DeletionVectorUtils
+import com.databricks.sql.transaction.tahoe.sources.DeltaSQLConf
+import com.nvidia.spark.rapids.delta.{UpdateCommandEdgeMeta, UpdateCommandMeta}
+
+object UpdateCommandMetaShim {
+  def tagForGpu(meta: UpdateCommandMeta): Unit = {
+    val deltaLog = Option(meta.updateCmd.tahoeFileIndex).map(_.deltaLog).orNull
+    val dvFeatureEnabled = deltaLog == null ||
+      DeletionVectorUtils.deletionVectorsWritable(deltaLog.unsafeVolatileSnapshot)
+
+    if (dvFeatureEnabled && meta.updateCmd.conf.getConf(
+      DeltaSQLConf.UPDATE_USE_PERSISTENT_DELETION_VECTORS)) {
```

Comment:
it should check Delta metadata too. Otherwise we disable it for tables that disable Dvs, no? https://github.com/delta-io/delta/blob/ce8bee4bfac21efdf9ab37e99311c2fc4808616b/spark/src/main/scala/org/apache/spark/sql/delta/commands/DeletionVectorUtils.scala#L74

Reply:
I just confirmed that `DeletionVectorUtils.deletionVectorsWritable` returns true only if deletionVectors are enabled in the protocol and the metadata.

Reply:
But where does this PR do it? This is what we want for the plugin too

Reply:
Line 27

Reply:
This alone was not sufficient to solve tightBounds. We need to handle to create a common method somewhere that handle newProtocol/newMetadata https://github.com/NVIDIA/spark-rapids/pull/12451/files#diff-be51435857c8be1c1fb38bfd9f0b0a0e9d6ad16982d45082f549d7b85a2f36f6R95

Reply:
The referenced code was only looking at the protocol. The method `DeletionVectorUtils.deletionVectorsWritable` looks at the protocol and metadata. I can share more offline

Reply:
Sorry, I missed the 'new' in your comment. Do we have access to the transaction at the planning time when the tagging is done? I am looking

Reply:
I have confirmed that we don't have access to the `transaction` object at the time of planning. The `TahoeFileIndex` object should have the updated snapshot after a Delta scan, and it's OK to use this to determine if the deletion vectors should be used.

Reply:
DeletionVectorUtils.deletionVectorsWritable already seems to check the feature. 

Is there an advantage of going into deltaLog.unsafeVolatileSnapshot
Should not this check just be
```
    if (DeletionVectorUtils.deletionVectorsWritable(meta.updateCmd.tahoeFileIndex)) {
      // https://github.com/NVIDIA/spark-rapids/issues/8654
      meta.willNotWorkOnGpu("Deletion vector writes are not supported on GPU")
    }

```

Reply:
Since we are not going to support DV this release we can set enableDeletionVectors to false and add a few fallback tests

Reply:
I am looking into this test failure. We are falling back on writing into Delta. This is probably due to the `UpdateCommandEdge` now falling back but I am still looking into how this test which explicitly sets the `enableDeletionVectors=True` was passing on 13.3

Reply:
I understand it now. In previous versions of Databricks we were writing UpdateCommand on the GPU regardless of the deletion vector status which is why the test was passing. This PR explicitly falls back to the CPU, just like the other tests where we xfail if the deletion vectors are enabled, we should do the same. We already have a test to check if `UpdateCommand` falls back to the CPU so I don't think we need another one.